src/lib/proof-system/zkprogram.ts

```diff
@@ -729,21 +721,21 @@ function picklesRuleFromFunction(
     assert(!(inProver && argsWithoutPublicInput === undefined));
     let finalArgs = [];
     let proofs: {
-      proofInstance: ProofBase<any, any>;
-      classReference: Subclass<typeof ProofBase<any, any>>;
+      Proof: Subclass<typeof ProofBase<any, any>>;
+      proof: ProofBase<any, any>;
     }[] = [];
     let previousStatements: Pickles.Statement<FieldVar>[] = [];
     for (let i = 0; i < args.length; i++) {
-      let { type, isProof } = args[i];
+      let type = args[i];
       try {
         let value = Provable.witness(type, () => {
-          return argsWithoutPublicInput?.[i] ?? emptyValue(type);
+          return argsWithoutPublicInput?.[i] ?? ProvableType.synthesize(type);
         });
         finalArgs[i] = value;
-        if (isProof) {
-          let Proof = type as Subclass<typeof ProofBase<any, any>>;
-          let proof = value as ProofBase<any, any>;
-          proofs.push({ proofInstance: proof, classReference: Proof });
+
```

Comment:
I'm just curious why we don't check `if(isProof){` here anymore?

Reply:
now, every method argument can contain multiple proofs, so I iterate over all the proofs we get from `extractProofs()`, rather than just test if the top-level method argument is a proof